client/src/components/SignInForm.tsx

```diff
@@ -0,0 +1,88 @@
+import { type FormEvent, useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useUser } from './useUser';
+
+type AuthData = {
+  user: User;
+  token: string;
+};
+
+/**
+ * Form that signs in a user.
+ */
+export function SignInForm() {
+  const { handleSignIn } = useUser();
+  const [isLoading, setIsLoading] = useState(false);
+  const { user } = useUser();
```

Comment:
You are using `useUser` on both lines 14 & 16, why not destructure `handleSignIn` and `user` on the same line?